projects/homework 5/postfix_calculator/calculate.c

```diff
@@ -0,0 +1,80 @@
+#include "../stack/stack.h"
+#include "calculate.h"
+#include <stdlib.h>
+#include <string.h>
+
+Stack* selectOperation(Stack* stack, char operation, int* errorCode)
+{
+    int firstDigit = top(stack, errorCode);
+    stack = pop(stack, &firstDigit);
+    int secondDigit = top(stack, errorCode);
+    if (*errorCode == 1)
+    {
+        return NULL;
+    }
+    stack = pop(stack, &secondDigit);
+    int result = 0;
+    switch (operation)
+    {
+    case '+':
+        result = secondDigit + firstDigit;
+        break;
+    case '-':
+        result = secondDigit - firstDigit;
+        break;
+    case '*':
+        result = secondDigit * firstDigit;
+        break;
+    case '/':
+        if (firstDigit != 0)
+        {
+            result = secondDigit / firstDigit;
+        }
+        else
+        {
+            *errorCode = 3;
+            return deleteStack(stack);
+        }
+        break;
+    }
+    return push(stack, result);
+}
+
+int calculate(char* string, int* errorCode)
```

Comment:
```suggestion
int calculate(const char* string, int* errorCode)
```
Этим мы явно говорим, что не собираемся менять саму строку, и это даёт возможность передавать сюда строковые литералы без предупреждений компилятора.